packages/admin/admin/src/dataGrid/useDataGridRemote.tsx

```diff
@@ -7,9 +7,11 @@ import { useHistory, useLocation } from "react-router";
 export function useDataGridRemote({
     queryParamsPrefix = "",
     pageSize: initialPageSize = 20,
+    initialSort,
 }: {
     queryParamsPrefix?: string;
     pageSize?: number;
+    initialSort?: Array<{ field: string; sort: GridSortDirection }>;
```

Comment:
no because here we need the DataGrid specific types not the Graphql types. The `muiGridSortToGql` function converts the direction to the Graphql type.

Reply:
It should be possible to either specify the `...SortField` or the `...QueryVariables` as a generic though.

Reply:
useDataGridRemote works on mui level - unrelated to any graphql api. And mui types field as string (see https://github.com/mui/mui-x/blob/v5.x/packages/grid/x-data-grid/src/models/gridSortModel.ts#L34)

That's why I'd also say it's perfect that way.